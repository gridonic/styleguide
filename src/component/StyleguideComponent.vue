<template>
    <div class="styleguide-component">
        <h2 class="name">
            {{ name }}
        </h2>
        <div class="examples">
            <h3 class="title">
                Example
            </h3>
            <div
                v-for="example in examples"
                :key="example"
            >
                <!-- eslint-disable-next-line -->
                <span v-html="example"/>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        name: {
            type: String,
            required: true,
        },
        examples: {
            type: Array,
            default: () => [],
        },
    },
};
</script>

<style lang="scss" scoped>
    .styleguide-component {
        > .name {
            margin: 0;
        }

        > .examples {
            > .title {
                text-transform: uppercase;
                color: #E7E3D7;
                font-weight: normal;
            }
        }
    }
</style>
