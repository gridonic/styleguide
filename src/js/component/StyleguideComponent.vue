<template>
    <div class="styleguide-component">
        <div class="information -typography-styleguide">
            <h2 class="name">
                {{ name }}
            </h2>
        </div>
        <div class="exampletitle -typography-styleguide">
            <h3>
                Example
            </h3>
        </div>
        <div class="examples">
            <div
                class="example"
                v-for="example in examples"
                :key="example"
            >
                <!-- eslint-disable-next-line -->
                <span v-html="example"/>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        name: {
            type: String,
            required: true,
        },
        examples: {
            type: Array,
            default: () => [],
        },
    },
};
</script>

<style lang="scss" scoped>
    .styleguide-component {
        > .information {
            > .name {
                margin: 0;
            }
        }

        > .exampletitle {
            margin-top: 20px;
        }

        > .examples {
            margin-top: 10px;

            display: grid;
            grid-auto-flow: row dense;
            grid-gap: 15px;
        }
    }
</style>
