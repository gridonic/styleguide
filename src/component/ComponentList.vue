<template>
    <div class="component-list">
        <div
            v-for="component in components"
            :key="component.name"
            class="component"
        >
            <styleguide-component v-bind="component" />
        </div>
    </div>
</template>

<script>
import StyleguideComponent from '@/component/StyleguideComponent';
import { mapGetters } from 'vuex';

export default {
    name: 'ComponentList',
    components: {
        StyleguideComponent,
    },
    computed: {
        ...mapGetters([
            'components',
        ]),
    },
};
</script>

<style lang="scss" scoped>
    .component-list {
        display: grid;
        grid-row-gap: 1em;
        grid-auto-flow: row dense;

        > .component {
        }
    }
</style>
