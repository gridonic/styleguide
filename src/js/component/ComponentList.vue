<template>
    <div class="component-list">
        <div
            v-for="component in components"
            :key="component.name"
            class="component"
        >
            <styleguide-component
                v-bind="component"
                :iframe="meta.iframe"
            />
        </div>
    </div>
</template>

<script>
import StyleguideComponent from '@/js/component/StyleguideComponent';
import { mapGetters } from 'vuex';

export default {
    name: 'ComponentList',
    components: {
        StyleguideComponent,
    },
    computed: {
        ...mapGetters([
            'components',
            'meta',
        ]),
    },
};
</script>

<style lang="scss" scoped>
    .component-list {
        display: grid;
        grid-row-gap: 2em;
        grid-auto-flow: row dense;

        > .component {
        }
    }
</style>
