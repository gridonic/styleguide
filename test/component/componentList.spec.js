import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import ComponentList from '@/js/component/ComponentList.vue';
import StyleguideComponent from '@/js/component/StyleguideComponent.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
    getters: {
        meta() {
            return {};
        },
        components() {
            return [
                { name: 'Component A' },
                { name: 'Component B' },
                { name: 'Component C' },
            ];
        },
    },
});


describe('ComponentList', () => {
    const build = () => {
        const wrapper = shallowMount(ComponentList, { localVue, store });
        return {
            wrapper,
            styleguideComponents: () => wrapper.findAll(StyleguideComponent),
        };
    };


    it('renders the component', () => {
        const { wrapper } = build();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders three StyleguideComponents', () => {
        const { styleguideComponents } = build();

        expect(styleguideComponents().length).toBe(3);
    });

    it('should pass correct data to each styleguide component', () => {
        const { styleguideComponents } = build();

        const components = styleguideComponents();

        expect(components.at(0).props('name')).toBe('Component A');
        expect(components.at(1).props('name')).toBe('Component B');
        expect(components.at(2).props('name')).toBe('Component C');
    });
});
