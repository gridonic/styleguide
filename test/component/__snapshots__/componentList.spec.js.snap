// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentList renders the component 1`] = `"<div class=\\"component-list\\"><div class=\\"component\\"><styleguide-component-stub name=\\"Component A\\" examples=\\"\\"></styleguide-component-stub></div><div class=\\"component\\"><styleguide-component-stub name=\\"Component B\\" examples=\\"\\"></styleguide-component-stub></div><div class=\\"component\\"><styleguide-component-stub name=\\"Component C\\" examples=\\"\\"></styleguide-component-stub></div></div>"`;
